From 9370ec9985b8fa528eb56072a9aaca34ed9ea57a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.net>
Date: Fri, 2 Nov 2012 23:29:25 +0000
Subject: [PATCH] backend: really use scaletempo instead of soundpitch plugin

The soundpitch plugin is irreparably broken, and the scaletempo
offers the same features without the bugs, or the need to change
the pitch ratio ourselves.

https://bugzilla.gnome.org/show_bug.cgi?id=686391
---
 src/backend/bacon-video-widget.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/backend/bacon-video-widget.c b/src/backend/bacon-video-widget.c
index 99bc075..0b1ee39 100644
--- a/src/backend/bacon-video-widget.c
+++ b/src/backend/bacon-video-widget.c
@@ -5963,7 +5963,7 @@ bacon_video_widget_initable_init (GInitable     *initable,
 
   /* Instantiate all the fallible plugins */
   bvw->priv->play = element_make_or_warn ("playbin", "play");
-  bvw->priv->audio_pitchcontrol = element_make_or_warn ("pitch", "scaletempo");
+  bvw->priv->audio_pitchcontrol = element_make_or_warn ("scaletempo", "scaletempo");
   video_sink = element_make_or_warn ("cluttersink", "video-sink");
   audio_sink = element_make_or_warn ("autoaudiosink", "audio-sink");
 
@@ -6150,7 +6150,6 @@ bacon_video_widget_set_rate (BaconVideoWidget *bvw,
   GstEvent *event;
   gboolean retval = FALSE;
   gint64 cur;
-  gfloat pitch, ratio;
 
   g_return_val_if_fail (BACON_IS_VIDEO_WIDGET (bvw), FALSE);
   g_return_val_if_fail (GST_IS_ELEMENT (bvw->priv->play), FALSE);
@@ -6160,7 +6159,6 @@ bacon_video_widget_set_rate (BaconVideoWidget *bvw,
 	return TRUE;
   if (new_rate >= 2.0)
 	return TRUE;
-  ratio = new_rate / bvw->priv->rate;
 
   if (gst_element_query_position (bvw->priv->play, GST_FORMAT_TIME, &cur)) {
     GST_DEBUG ("Setting new rate at %"G_GINT64_FORMAT"", cur);
-- 
1.8.0

